src/components/link/link.component.tsx

```diff
@@ -85,25 +84,6 @@ export const Link = React.forwardRef<
     const [hasFocus, setHasFocus] = useState(false);
     const l = useLocale();
     const { inMenu } = useContext(MenuContext);
-    const handleOnKeyDown = (
```

Comment:
**praise**: Good work Marc, this function does appear to be the culprit of the unintended behaviour, as we're calling both `onClick` and `onKeyDown` resulting in the unintended "double click" behaviour